packages/bpk-component-barchart/src/BpkBarchart.js

```diff
@@ -45,17 +47,88 @@ const getClassName = cssModules(STYLES);
 const spacing = remToPx(spacingXs);
 const lineHeight = remToPx(lineHeightSm);
 
-const getMaxYValue = (dataPoints, yScaleDataKey, outlierPercentage) => {
+const getMaxYValue = (dataPoints: Function, outlierPercentage: ?number) => {
   const meanValue = dataPoints.reduce((d, t) => d + t, 0) / dataPoints.length;
   const maxYValue = Math.max(...dataPoints);
 
-  return outlierPercentage !== null
+  return outlierPercentage
     ? Math.min(maxYValue, meanValue * (outlierPercentage / 100) + meanValue)
     : maxYValue;
 };
 
-class BpkBarchart extends Component {
-  constructor(props) {
+type Props = {
+  data: any,
+  xScaleDataKey: string,
+  yScaleDataKey: string,
+  xAxisLabel: string,
+  yAxisLabel: string,
+  initialWidth: number,
+  initialHeight: number,
+  className: ?string,
+  leadingScrollIndicatorClassName: ?string,
+  trailingScrollIndicatorClassName: ?string,
+  outlierPercentage: ?number,
+  showGridlines: boolean,
+  xAxisMargin: number,
+  xAxisTickValue: Function,
+  xAxisTickOffset: number,
+  xAxisTickEvery: number,
+  yAxisMargin: number,
+  yAxisTickValue: Function,
+  yAxisNumTicks: ?number,
+  yAxisDomain: Array<?number>,
+  onBarClick: ?Function,
+  onBarHover: ?Function,
+  onBarFocus: ?Function,
+  getBarLabel: Function,
+  getBarSelection: Function,
+  BarComponent: Function,
+  disableDataTable: boolean,
+};
+
+type State = {
+  width: number,
+  height: number,
+};
+
+class BpkBarchart extends Component<Props, State> {
+  xScale: any;
+
+  yScale: any;
+
+  onWindowResize: Function;
+
+  svgEl: any;
+
+  static defaultProps = {
+    data: null,
+    className: null,
+    leadingScrollIndicatorClassName: null,
+    trailingScrollIndicatorClassName: null,
+    outlierPercentage: null,
+    showGridlines: false,
+    xAxisMargin: 2 * (lineHeight + spacing),
+    xAxisTickValue: identity,
+    xAxisTickOffset: 0,
+    xAxisTickEvery: 1,
+    yAxisMargin: 4 * lineHeight + spacing,
+    yAxisTickValue: identity,
+    yAxisNumTicks: null,
+    yAxisDomain: [null, null],
+    onBarClick: null,
+    onBarHover: null,
+    onBarFocus: null,
+    getBarLabel: (
+      point: Array<*>,
```

Comment:
The `*` seems suspect, can we not use something more specific? This may also tie into my suggestion from earlier about creating a `PointType`.